### PR TITLE
Improved quite a few comments

### PR DESCRIPTION
**Pull request by GabrielRavier**
Just added a bunch of comments documenting problems and possible errors in the code

### CODE REVIEW

File: src/MycParam.cpp [hunk 1]
```diff
@@ -135,7 +135,7 @@ void DamageMyChar(int damage)
 
 	// Lose a whimsical star
 	if (gMC.equip & 0x80 && gMC.star > 0)
-		gMC.star = (short)gMC.star - 1;	// Why the hell is it written this way?
+		gMC.star = (short)gMC.star - 1;	// For some reason, does a cast to short. Probably not accurate to the original source
```
Comment (Clownacy) [inline src/MycParam.cpp]: I dunno... there are plenty of other weird casts elsewhere in the code, like whenever a 'deg' variable is used. I think Pixel might have just been cautious about int size conversion and underflow/overflow.

File: src/NpcAct080.cpp [hunk 2]
```diff
@@ -1080,6 +1080,8 @@ void ActNpc088(NPCHAR *npc)
 		case 10:
 			if (++npc->act_wait > 100 && npc->act_wait % 6 == 1)
 			{
+				// The range of unsigned char is [0, 255] so these two values are implicitly truncated (Might be UB ?)
+				// TODO : Investigate on whether this is erroneous
```
Comment (Clownacy) [inline src/NpcAct080.cpp]: This has been fixed in e34f492418a6ec18c6b24303c31ca043f8e3e143

File: src/NpcAct160.cpp [hunk 3]
```diff
@@ -70,7 +70,7 @@ void ActNpc160(NPCHAR *npc)
 
 		case 3:
 			npc->damage = 20;
-			npc->damage = 0;	// Smart code
+			npc->damage = 0;	// Duplicate line
```
Comment (Clownacy) [inline src/NpcAct160.cpp]: I'd prefer the top line be commented instead, pointing out that it's overwritten by the line below.

File: src/NpcAct260.cpp [hunk 4]
```diff
@@ -1291,6 +1291,8 @@ void ActNpc268(NPCHAR *npc)
 		case 51:
 			if (++npc->act_wait > 30 && npc->act_wait % 4 == 1)
 			{
+				// The range of unsigned char is [0, 255] so these two values are implicitly truncated (Might be UB ?)
+				// TODO : Investigate on whether this is erroneous
```
Comment (Clownacy) [inline src/NpcAct260.cpp]: Fixed in 5026964d98a818cbbeecb40847aa240a669f65e5

File: src/MycParam.cpp (hunk 1 above)
Comment (GabrielRavier) [reply src/MycParam.cpp]: I tried to clarify more (discussing some possibilities) and changed the "Probably" to a more neutral "Might".

File: src/NpcAct080.cpp (hunk 2 above)
Comment (GabrielRavier) [reply src/NpcAct080.cpp]: Removed it

File: src/NpcAct260.cpp (hunk 4 above)
Comment (GabrielRavier) [reply src/NpcAct260.cpp]: Removed it

File: src/NpcAct160.cpp (hunk 3 above)
Comment (GabrielRavier) [reply src/NpcAct160.cpp]: Done

**Pull request merged**